workflows/api/loader/loader.py

```diff
@@ -67,62 +74,71 @@ async def process_workflow_completed(
             )
         await asyncio.gather(*loader_futures)
 
-    async def main(self) -> None:
-        """Waits for events and if a workflow completes, runs the loaders"""
-        print("Running listener")
-        while True:
-            for event in await self.bus.poll():
-                print("event", event, file=sys.stderr)
-                if isinstance(event, WorkflowStartedMessage):
-                    workflow_run = (
-                        await self.session.execute(
-                            select(WorkflowRun).where(WorkflowRun.execution_id == event.runner_id)
-                        )
-                    ).scalar_one_or_none()
-                    if workflow_run and workflow_run.status in [
-                        WorkflowRunStatus.CREATED,
-                        WorkflowRunStatus.PENDING,
-                        WorkflowRunStatus.STARTED,
-                    ]:
-                        workflow_run.status = WorkflowRunStatus.RUNNING
-                        await self.session.commit()
-
-                if isinstance(event, WorkflowSucceededMessage):
-                    _event: WorkflowSucceededMessage = event
-                    result = await self.session.execute(
-                        select(WorkflowRun)
-                        .options(
-                            joinedload(WorkflowRun.workflow_version).joinedload(WorkflowVersion.workflow),
-                            selectinload(WorkflowRun.entity_inputs),
+    def _bind_handle_message(self) -> Callable[[WorkflowStatusMessage], Awaitable[None]]:
+        async def handle_message(event: WorkflowStatusMessage) -> None:
+            print("event", event, file=sys.stderr)
+            if isinstance(event, WorkflowStartedMessage):
+                workflow_run = (
+                    await self.session.execute(select(WorkflowRun).where(WorkflowRun.execution_id == event.runner_id))
+                ).scalar_one_or_none()
```

Comment:
Right now with this one_or_none logic, if we don't find a matching workflow run we just ignore it and silently drop the message. Should we be raising an error here?

Reply:
No, this is intentional because we will be getting mngs messages

Reply:
ok makes sense, so the retry only happens if there is an error in the loaders

Reply:
If there is an error in the loaders or if for whatever reason the process is randomly killed. Loader errors may not really be retry-able unless there is a transient issue communicating with the entities service so for those failures it is more about pushing them into the dead letter queue than retrying.